Backpack/Gradient/Classes/BPKGradient.m

```diff
@@ -61,6 +61,15 @@ - (instancetype)initWithColors:(NSArray<UIColor *> *)colors
     return self;
 }
 
+- (instancetype)cloneWithNewDirection:(BPKGradientDirection)newDirection {
```

Comment:
Why do we need `cloneWithNewDirection`. Would it make more sense for consumers to just clone a gradient and set the direction on the resulting view?

Reply:
I am confused, can you share some code examples?

Reply:
For example, why can't a user do something like this:
```
let gradient = Gradient();
let someOtherGradient = gradient.clone()
someOtherGradient.directionOverride = GradientDirectionTopLeft
```

Reply:
I would expect the API to allow me to just set the direction on the gradient instance

Reply:
This lets you do that, but it prefers a readonly API which is much nicer than a mutable API as you suggested